src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceService.java

```diff
@@ -208,4 +225,109 @@ public String deleteControlledGcsBucket(
 
     return jobBuilder.submit();
   }
+
+  private void validateOnlySelfAssignmentForWorkspaceWriter(
+      ControlledResource controlledResource, AuthenticatedUserRequest userRequest) {
+    if (!controlledResource.getAccessScope().equals(AccessScopeType.ACCESS_SCOPE_PRIVATE)) {
+      // No need to handle SHARED resources
+      return;
+    }
+    final boolean userIsWriterOnWorkspace =
+        samService.isAuthorized(
+            userRequest.getRequiredToken(),
+            SamConstants.SAM_WORKSPACE_RESOURCE,
+            controlledResource.getWorkspaceId().toString(),
+            SamConstants.SAM_WORKSPACE_WRITE_ACTION);
+    if (userIsWriterOnWorkspace) {
+      // UserEmail field isn't always populated for requests going through Swagger, so get it the
+      // hard way
+      final String requestUserEmail = samService.getEmailFromToken(userRequest.getRequiredToken());
+
+      final boolean isAllowed =
+          controlledResource
+              .getAssignedUser()
+              .map(requestUserEmail::equals)
+              .orElseThrow(
+                  () ->
+                      new BadRequestException(
+                          "Assigned User field is required for private controlled resources."));
+      if (!isAllowed) {
+        throw new BadRequestException(
+            "Workspace Writer User may only assign a private controlled resource to themselves.");
+      }
+    }
+  }
+
+  /**
+   * Workspace READERs may only have READER permission on private controlled resources to which they are
+   * assigned. WRITERs may have either READER or WRITER permission, and OWNERS are unrestricted.
+   *
+   * @param controlledResource - controlled resource to check permission match on
+   * @param userRequest - current request
+   */
+  private void validateWorkspaceReadersOnlyHaveReaderRoleOnResources(
```

Comment:
Delete this check, because only writers and owners are creating resources anyway. (self-assignment only).